is_core/menu.py

```diff
@@ -43,12 +43,18 @@ def get_menu_items(self, items):
         for item in items:
             if isinstance(item, MenuItem):
                 menu_items.append(item)
-            else:
-                item = self.site._registry[item]
+            elif self.site.has_iscore(item):
+                item = self.site.get_iscore(item)
                 menu_item = item.get_menu_item(self.request, group)
                 if menu_item:
                     menu_items.append(menu_item)
+            else:
+                raise MenuError('IS Core or Menu item not found for name "%s"' % item)
         return menu_items
 
     def get_menu_structure(self):
-        return self.site._registry.keys()
+        return self.site.get_iscore_names()
+
+
+class MenuError(Exception):
```

Comment:
we have exceptions package for errors in django-is-core, but maybe ImproperlyConfigured error will be enought. There is no reason to create custom exception